src/pages/studyView/charts/barChart/CustomBinsModal.tsx

```diff
@@ -2,16 +2,28 @@ import * as React from 'react';
 import _ from 'lodash';
 import { ChartMeta, customBinsAreValid } from 'pages/studyView/StudyViewUtils';
 import autobind from 'autobind-decorator';
-import { observable, computed, makeObservable } from 'mobx';
+import { observable, computed, makeObservable, action } from 'mobx';
 import { observer } from 'mobx-react';
-import { Modal, Button } from 'react-bootstrap';
+import { Modal } from 'react-bootstrap';
+import LabeledCheckbox from 'shared/components/labeledCheckbox/LabeledCheckbox';
+import { GenerateBinsConfig } from 'cbioportal-ts-api-client/dist/generated/CBioPortalAPIInternal';
+import {
+    BinMethodOption,
+    StudyViewPageStore,
+} from 'pages/studyView/StudyViewPageStore';
 
 export type ICustomBinsProps = {
     show: boolean;
     onHide: () => void;
     chartMeta: ChartMeta;
     currentBins: number[];
-    updateCustomBins: (uniqueKey: string, bins: number[]) => void;
+    updateCustomBins: (
+        uniqueKey: string,
+        bins: number[],
+        binMethod: string,
```

Comment:
Should be of type GenomicDataBinFilter.binMethod

Reply:
Not working because GenomicDataBinFilter is a type